pyaedt/icepak.py

```diff
@@ -4625,3 +4625,266 @@ def _create_node_dict(self, default_dict):
+              ``"Function"`` key, two parameters are required: the value of the variable
+              at t=0 and the slope of the line. For the parameters required by each
+              ``"Function"`` key selection, see the Icepack documentation (). The parameters
+              must contain the units where needed.
+
+        boundary_name : str, optional
+            Name of the source boundary. The default is ``None``, in which case the
+            boundary is automatically generated.
+        external_temperature : str, dict or float, optional
+            String with the value and unit of the temperature for the heat transfer
+            coefficient. If a float value is specified, the ``"cel"`` unit is automatically
+            added.
+            For a transient assignment, a dictionary can be used as described for the
+            assignment_value argument. Temperature dependent assignment is not supported.
+            The default is ``"AmbientTemp"``.
+
+
+        Returns
+        -------
+        :class:`pyaedt.modules.Boundary.BoundaryObject`
+            Boundary object when successful or ``None`` when failed.
+
+        References
+        ----------
+
+        >>> oModule.AssignBlockBoundary
+
+        Examples
+        --------
+        >>> from pyaedt import Icepak
+        >>> ipk = Icepak()
+        >>> ipk.solution_type = "Transient"
+        >>> box = ipk.modeler.create_box([5, 5, 5], [1, 2, 3], "BlockBox5", "copper")
+        >>> box.solve_inside = False
+        >>> temp_dict = {"Type": "Transient", "Function": "Square Wave", "Values": ["1cel", "0s", "1s", "0.5s", "0cel"]}
+        >>> block = ipk.assign_hollow_block("BlockBox5", "Heat Transfer Coefficient", "1w_per_m2kel", "Test", temp_dict)
+        """
+        if self.modeler.get_object_from_name(object_name).solve_inside:
+            self.logger.add_error_message(
+                "Use ``assign_solid_block`` function with this object as" "``solve_inside`` is ``True``."
+            )
+            return None
+        if assignment_value == "Joule Heating" and assignment_type != "Total Power":
+            self.logger.add_error_message(
+                '``"Joule Heating"`` assignment is supported only if ``assignment_type``' 'is ``"Total Power"``.'
+            )
+            return None
+
+        assignment_dict = {
+            "Total Power": ["Fixed Heat", "Total Power"],
+            "Heat Flux": ["Fixed Heat", "Heat Flux"],
+            "Temperature": ["Fixed Temperature", "Fixed Temperature"],
+            "Heat Transfer Coefficient": ["Internal Conditions", "Heat Transfer Coefficient"],
+        }
+        thermal_condition = assignment_dict.get(assignment_type, None)
+        if thermal_condition is None:
+            self.logger.add_error_message(
+                'Available assignment_type are "Total Power", "Heat Flux",'
+                '"Temperature" and "Heat Transfer Coefficient".'
```

Comment:
```suggestion
                'Available assignment_type are "Total Power", "Heat Flux",'
                '"Temperature", and "Heat Transfer Coefficient".'
```
Error messages should use text when possible to be more readable. Consider using something like this: Valid options for assignment type are ....